GVFS/GVFS.Service/Handlers/RequestHandler.cs

```diff
@@ -0,0 +1,112 @@
+            if (string.IsNullOrWhiteSpace(message.Header))
+            {
+                return;
+            }
+
+            using (ITracer activity = this.tracer.StartActivity(message.Header, EventLevel.Informational, new EventMetadata { { nameof(request), request } }))
+            {
+                try
+                {
+                    this.HandleMessage(activity, message, connection);
+                }
+                catch (SerializationException ex)
+                {
+                    EventMetadata metadata = new EventMetadata();
+                    metadata.Add("Area", this.etwArea);
+                    metadata.Add("Header", message.Header);
+                    metadata.Add("Exception", ex.ToString());
+
+                    activity.RelatedError(metadata, $"Could not deserialize {this.requestDescription} request: {ex.Message}");
+                }
+            }
+        }
+
+        protected virtual void HandleMessage(
+            ITracer tracer,
+            NamedPipeMessages.Message message,
+            NamedPipeServer.Connection connection)
+        {
+            switch (message.Header)
+            {
+                case NamedPipeMessages.RegisterRepoRequest.Header:
+                    this.requestDescription = MountRequestDescription;
+                    NamedPipeMessages.RegisterRepoRequest mountRequest = NamedPipeMessages.RegisterRepoRequest.FromMessage(message);
+                    RegisterRepoHandler mountHandler = new RegisterRepoHandler(tracer, this.repoRegistry, connection, mountRequest);
+                    mountHandler.Run();
+
+                    break;
+
+                case NamedPipeMessages.UnregisterRepoRequest.Header:
+                    this.requestDescription = UnmountRequestDescription;
+                    NamedPipeMessages.UnregisterRepoRequest unmountRequest = NamedPipeMessages.UnregisterRepoRequest.FromMessage(message);
+                    UnregisterRepoHandler unmountHandler = new UnregisterRepoHandler(tracer, this.repoRegistry, connection, unmountRequest);
+                    unmountHandler.Run();
+
+                    break;
+
+                case NamedPipeMessages.GetActiveRepoListRequest.Header:
+                    this.requestDescription = RepoListRequestDescription;
+                    NamedPipeMessages.GetActiveRepoListRequest repoListRequest = NamedPipeMessages.GetActiveRepoListRequest.FromMessage(message);
+                    GetActiveRepoListHandler excludeHandler = new GetActiveRepoListHandler(tracer, this.repoRegistry, connection, repoListRequest);
+                    excludeHandler.Run();
+
+                    break;
+
+                case NamedPipeMessages.EnableAndAttachProjFSRequest.Header:
+
+                    // This request is ignored on non Windows platforms.
+                    NamedPipeMessages.EnableAndAttachProjFSRequest.Response response = new NamedPipeMessages.EnableAndAttachProjFSRequest.Response();

```

Comment:
Related to comment in RequestHandler.Windows.cs, we should still set `this.requestDescription` so that it can be logged properly (if needed) in `HandleRequest`.